apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -206,8 +251,13 @@ namespace MWRender
     {
         resourceSystem->getSceneManager()->setParticleSystemMask(MWRender::Mask_ParticleSystem);
         resourceSystem->getSceneManager()->setShaderPath(resourcePath + "/shaders");
-        // Shadows and radial fog have problems with fixed-function mode
-        bool forceShaders = Settings::Manager::getBool("radial fog", "Shaders") || Settings::Manager::getBool("force shaders", "Shaders") || Settings::Manager::getBool("enable shadows", "Shadows");
+        // Shadows and radial fog have problems with fixed-function mode.
+        // Groundcover fading and animation are implemented via shader too.
+        bool forceShaders =
+            Settings::Manager::getBool("radial fog", "Shaders") ||
+            Settings::Manager::getBool("force shaders", "Shaders") ||
+            Settings::Manager::getBool("enable shadows", "Shadows") ||
+            Settings::Manager::getBool("enabled", "Groundcover");
```

Comment:
We shouldn't need to force shaders for *everything* just because groundcover needs them. Maybe it makes more sense to give `recreateShaders` and `createShaderVisitor` an extra parameter that forces shaders for that specific subgraph. If we go for that approach and you're changing that code anyway, it might be a good idea to skip the `recreateShaders` call and go directly for `createShaderVisitor` and `traverse` so it can happen with `allowedToModifyStateSets` enabled. It looks like they've already been deep copied, so there's no advantage to copying them a second time.